gui-js/apps/minsky-electron/src/app/backend/backend.ts

```diff
@@ -0,0 +1,120 @@
+import {  importCSVerrorMessage, } from '@minsky/shared';
+import { Utility } from '../utility';
+import { dialog } from 'electron';
+import * as elog from 'electron-log';
+import { join } from 'path';
+
+const JSON5 = require('json5');
+
+const addonDir = Utility.isPackaged()
+  ? '../node-addons'
+  : '../../node-addons';
+
+var log=elog? elog: console;
+if (!Utility.isDevelopmentMode()) { //clobber logging in production
+  log.info=function(...args: any[]){};
+};
+
+/** returns true if RESTService call is logged in development mode */
+function logFilter(c: string) {
+  const logFilter=["mouseMove$", "requestRedraw$"];
+  for (var i in logFilter)
+    if (c.match(logFilter[i])) return false;
+  return true;
+}
+
+/** REST Service addon */
+export var restService = null;
+try {
+  restService = require('bindings')(join(addonDir, 'minskyRESTService.node'));
+} catch (error) {
+  log.error(error);
+}
+
+var errorDialog=(message: string)=>{};
+/** enable popup display of exception thrown by backend */
+export function setErrorDialog(f: (message: string)=>void) {errorDialog=f}
+
+/** core function to call into C++ object heirarachy */
+export function backend(command: string, ...args: any[]) {
+  if (!command) {
+    log.error('backend called without any command');
+    return {};
+  }
+  if (!restService) {
+    log.error('Rest Service not ready');
+    return {};
+  }
+  try {
+    var arg='';
+    if (args.length>0)
```

Comment:
I noticed there are some formatting peculiarities that you're partial to, such as the lack of spaces around the > symbol here... I'm normally fine with that, but it might be worth it for us to settle on a set of standards and apply them consistently across the TS code base... maybe even automatically. Currently there isn't much consistency...

Reply:
If we do that, the only thing I ask is to settle on a formatting program to do it automatically. I don't want to think about whether to put spaces in or not when writing code. And I certainly don't care about it when reading code.